File: Java/q1040.java
// Média 3

import java.util.Scanner;
import java.text.DecimalFormat;

public class Main {
    public static void main(String[] args) {
      DecimalFormat df = new DecimalFormat("0.0");
      Scanner input = new Scanner(System.in);
      double n1 = input.nextDouble();
      double n2 = input.nextDouble();
      double n3 = input.nextDouble();
      double n4 = input.nextDouble();
      double media = (n1*2 + n2*3 + n3*4 + n4)/10;
    System.out.println("Media: " + df.format(media));
    				if (media >= 5.0 && media < 7.0){
    								System.out.println("Aluno em exame.");
    								double notaExame = input.nextDouble();
    								double mediaFinal = (notaExame + media)/2;
    								System.out.println("Nota do exame: " + df.format(notaExame));
    								if (notaExame >= 5.0){
    												System.out.println("Aluno aprovado.");
       } else {
       				System.out.println("Aluno reprovado.");
       }
        
       System.out.println("Media final: " + df.format(mediaFinal));
    				} else if(media >= 7.0) {
    						  System.out.println("Aluno aprovado.");
    				} else {
    						  System.out.println("Aluno reprovado.");
    		}
    }
}

File: Cpp/q1040.cpp
#include <iostream>
#include <iomanip>
using namespace std;

int main(){
    
    float n1, n2, n3, n4, ne, media, mf;
    cin >> n1 >> n2 >> n3 >> n4 >> ne;
    
    media = ((n1*2)+(n2*3)+(n3*4)+(n4*1))/10;
    cout << fixed << setprecision(1);
    cout << "Media: " << media << endl;

    if(media >= 7.0){
        cout << "Aluno aprovado.\n";
    } else if(media >= 5.0 && media <= 6.9){
        cout << "Aluno em exame.\n";
        cout << "Nota do exame: " << ne << endl;
        mf = (ne + media)/2;
        
        if(mf >= 5.0){
            cout << "Aluno aprovado.\n";
        } else {
            cout << "Aluno reprovado.\n";
        }

        cout << "Media final: " << mf << endl;
  
    } else {
        cout << "Aluno reprovado.\n";
    }

    return 0;
}